visualization/app/codeCharta/ui/artificialIntelligence/artificialIntelligence.component.ts

```diff
@@ -96,8 +86,33 @@ export class ArtificialIntelligenceController
 		}
 	}
 
-	applyCustomConfig(configId: string) {
-		CustomConfigHelper.applyCustomConfig(configId, this.storeService, this.threeCameraService, this.threeOrbitControlsService)
+	applySuspiciousMetric(metric: MetricSuggestionParameters, isOutlier: boolean) {
+		//TODO: Improve method
+		if (this._viewModel.suspiciousMetricSuggestionLinks.includes(metric)) {
+			const { mapColors } = this.storeService.getState().appSettings
+			const colorRange: ColorRange = {
+				from: metric.from,
+				to: metric.to,
+				max: 0,
+				min: 0
+			}
+
+			if (metric.isOutlier === isOutlier) {
+				mapColors.positive = "#ffffff"
+				mapColors.neutral = "#ffffff"
+				mapColors.negative = "#A900C0"
```

Comment:
This will mutate the state, since mapColors is a reference to an object inside the state.
The state should only be changed through dispatching an action. Instead, create a new object by spreading the mapColors from the state inside it, mutate the new object and dispatch it.

```
const mapColors = { ...this.storeService.getState().appSettings.mapColors }
```